x-pack/plugins/canvas/types/element.d.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+declare interface CanvasElementConfig {
```

Comment:
How are other typescript projects handling the filenames for types? This is exporting 2 types, and the filename matches neither. It doesn't seem like there's a [rule in the styleguide](https://github.com/elastic/kibana/blob/master/style_guides/architecture_style_guide.md), just curious if maybe it's not documented or if maybe we don't have one.

Reply:
Right... I created this file to contain types for anything related to `element`... rather than create dozens of `.d.ts` files for various files.  Since `.d.ts` files are never imported directly, the current "import all in an index.js" file approach doesn't apply.

@Stacey-Gammon - do you have a pattern in Kibana for this kind of thing?

Reply:
This was addressed in an update, created `types.ts` for importing types in each file.